### ISSUE TITLE
[BUG] Unable to config the `scalarsOverrides` or  `externalResolvers` property to get the custom resolver for `Date` and `DateTime` scalar types from `@neo4j/graphql` library

### ISSUE THREAD

**Issue by ScorpyG** [eddeee888/graphql-code-generator-plugins#331]
I'm having trouble with getting the codegen tool to work with Neo4jGraphQL... I have an issue with the scalar types (Date/DateTime). I'm aware that the Neo4j graphql library has its implementation for those scalar types that provide convenience to get stuff going (but I want to modularize my schemas) and using the codegen tool to stitch my schema together also generates the typings.

My general understanding of the issue is the graphql-codegen/cli package doesn't understand the Neo4j GraphQL scalars implementations and ends up causing errors when trying to generate the types. If try to manually define the type in the schema the tool will be able to compile and generate the type successfully but the apollo-graphql server would complain about the duplicated type that already exists in the schema.

Here is my setup so far:
package.json
```
dependencies: 
- @neo4j/graphql
- @apollo/server
- neo4j-driver
- @graphql-codegen/cli
- @eddeee888/gcg-typescript-resolver-files
- ...
```

src/schema/**.gql
```
type User {
  id: ID! @id
  birthday: Date!
  ...
  createdAt: DateTime @timestamp(operation: [CREATE])
}
```

codegen.ts
```
const config: CodegenConfig = {
  schema: '**/schema.gql',
  generates: {
    'src/graphql': defineConfig({
      // scalarsModule: false,
      scalarsOverrides: {
        Date: {
          resolver: '@neo4j/graphql#Date',
        },
        DateTime: {
          resolver: '@neo4j/graphql#DateTime',
        },
      },
    }),
  },
  hooks: { afterAllFileWrite: 'prettier --write' },
};

export default config;
```

<img width="461" alt="Screenshot 2024-09-08 at 8 42 34 PM" src="https://github.com/user-attachments/assets/88064cad-7e63-4285-8f96-0be9e3e115c1">

**Comment by eddeee888** [comment]
Hi @ScorpyG,

I believe this Codegen error generally happens because there's no `Date` scalar being declared in your schema files.

> If try to manually define the type in the schema the tool will be able to compile and generate the type successfully but the apollo-graphql server would complain about the duplicated type that already exists in the schema.

This leads me to believe something (probably Neo4j itself) adds the scalar declaration at runtime. How are you wiring up the type defs into Apollo Server?

One approach I can think of is to add a dummy schema file with `scalar Date` so that GraphQL codegen can generate the types, but do not include this file as type defs when passing to the schema. I haven't tested it so would be good to see how you are using your type defs so I can help further 🙂

**Comment by ScorpyG** [comment]
Hey @eddeee888, thank you for the response. Here is how I configured the typedefs on the Apollo server, following your suggestion of adding the dummy scalars `Date` and `DateTime` to the schema.

I simply imported the generated typedefs `graphql/typeDefs.generated` into my index.ts file.

![Screenshot 2024-09-09 at 7 58 17 AM](https://github.com/user-attachments/assets/f4f1da61-0ade-44be-be40-3a6086d5a09b)

I added both the scalar types `Date` & `DateTime` which I'm having issue with into the `base/schema.gql` file. 
![Screenshot 2024-09-09 at 7 59 11 AM](https://github.com/user-attachments/assets/7f3e1fb8-cdf8-4304-93f2-b64f669f064d)
